app/controllers/admin/patient_imports_controller.rb

```diff
@@ -0,0 +1,25 @@
+class Admin::PatientImportsController < AdminController
+  include FileUploadable
+
+  def new
+    authorize { current_admin.power_user? }
+  end
+
+  def create
+    authorize { current_admin.power_user? }
+
+    facility = Facility.find(params[:facility_id])
+
+    data = read_xlsx_or_csv_file(params[:patient_import_file])
+    params = PatientImport::SpreadsheetTransformer.call(data, facility: facility)
+    errors = PatientImport::Validator.new(params).errors
+
+    if errors.values.flatten.any?
```

Comment:
Will be neater if we could expose this from the validator:

`PatientImport::Validator.new(params).valid?`

Reply:
Oof, I like that a log.